c7n/resources/elasticsearch.py

```diff
@@ -92,6 +93,28 @@ def get_dimensions(self, resource):
                  'Value': resource['DomainName']}]
 
 
+@ElasticSearchDomain.filter_registry.register('kms-key')
+class KmsFilter(KmsRelatedFilter):
+    """
+    Filter a resource by its associcated kms key and optionally the aliasname
+    of the kms key by using 'c7n:AliasName'
+
+    :example:
+
+    .. code-block:: yaml
+
+        policies:
+          - name: elasticsearch-kms-key
+            resource: elasticsearch
```

Comment:
We prefer `resource: aws.elasticsearch` for best practices to reflect in the doc